plugins/chat/app/models/chat/direct_message_channel.rb

```diff
@@ -28,12 +28,13 @@ def generate_auto_slug
       self.slug.blank?
     end
 
-    def leave(user)
+    def remove(user)
```

Comment:
Why this change? I think it can break the `UnfollowChannel` service.

Reply:
There's an alias below. It brings this child-class to parity with `Channel`:

https://github.com/discourse/discourse/blob/b8a4b11cbba2cd3ef7d294e25b4ca35130bbf5fb/plugins/chat/app/models/chat/channel.rb#L143-L146

Reply:
That was done on purpose because for a category channel `#remove` and `#leave` follow the same logic. But for a DM, the `#leave` logic was different (we have an `UnfollowChannel` and a `LeaveChannel` services that don’t have the exact same logic, here’s when the change was introduced: https://github.com/discourse/discourse/commit/af6788fd331291d7205c2c27bf8b9281704124a7).

Reply:
Hm. When I was working on this, I added this because `Channel#remove` does not work correctly for a `DirectMessageChannel`. Was it intended to?

Reply:
I’m not sure either, but it seems `#remove` is used to just unfollow whereas `#leave` is used to leave a channel (but the behavior for a non-group DM has no difference between `#leave` and `#remove`).